components/gallery/GalleryItemAction/GalleryItemActionType/useGalleryItemAction.ts

```diff
@@ -0,0 +1,157 @@
+import { createInteraction } from '@kodadot1/minimark'
+import { Interaction } from '@kodadot1/minimark'
+import { checkAddress, isAddress } from '@polkadot/util-crypto'
+
+import { dangerMessage, infoMessage } from '@/utils/notification'
+import { bsxParamResolver, getApiCall } from '@/utils/gallery/abstractCalls'
+import { tokenIdToRoute } from '@/components/unique/utils'
+import { ss58Of } from '@/utils/config/chain.config'
+import correctFormat from '@/utils/ss58Format'
+
+type ActionList = {
+  interaction: Interaction.LIST
+  urlPrefix: string
+  price: string
+  nftId: string
+  successMessage?: string
+  errorMessage?: string
+}
+
+type ActionSend = {
+  interaction: Interaction.SEND
+  urlPrefix: string
+  tokenId: string
+  address: string
+  nftId: string
+  successMessage?: string
+  errorMessage?: string
+}
+
+type Actions = ActionList | ActionSend
+
+function checkBeforeList(item: ActionList) {
+  const meta = item.price
+
+  if (!meta) {
+    dangerMessage('Price is not valid')
+    return false
+  }
+
+  return true
+}
+
+function constructTransactionList(item: ActionList, api, executeTransaction) {
+  const meta = item.price
+  checkBeforeList(item)
+
+  if (item.urlPrefix === 'rmrk') {
+    executeTransaction({
+      cb: api.tx.system.remark,
+      arg: [createInteraction(Interaction.LIST, '1.0.0', item.nftId, meta)],
+      successMessage: item.successMessage,
+      errorMessage: item.errorMessage,
+    })
+  }
+
+  if (item.urlPrefix === 'snek' || item.urlPrefix === 'bsx') {
+    executeTransaction({
+      cb: getApiCall(api, item.urlPrefix, Interaction.LIST),
+      arg: bsxParamResolver(item.nftId, Interaction.LIST, meta),
+      successMessage: item.successMessage,
+      errorMessage: item.errorMessage,
+    })
+  }
+}
+
+function checkBeforeSend(item: ActionSend) {
+  const [, err] = checkAddress(
+    item.address,
+    correctFormat(ss58Of(item.urlPrefix))
+  )
+
+  if (!isAddress(item.address)) {
+    dangerMessage('Invalid address')
+    return
+  }
+
+  if (err) {
+    dangerMessage(err)
+    return
+  }
+}
+
+function constructTransactionSend(item: ActionSend, api, executeTransaction) {
+  const { id, item: token } = tokenIdToRoute(item.tokenId)
+  checkBeforeSend(item)
+
+  if (item.urlPrefix === 'rmrk') {
```

Comment:
using `switch & case` pattern to replace if/else would be better.

Reply:
IMO there is one main difference to `if/else`, `switch & case` would be easy to extend more option cases. 
By the way, object literal would also be a good option, that was an implement of `Strategy Pattern`.

Reply:
I think we should wrap it into SDK 
- https://github.com/kodadot/packages/issues/89